### DISCUSSION TITLE
Developer guide

### DISCUSSION THREAD

**Discussion by jolo-dev** [supabase/supabase#9324] [Questions]
Hi there,
I would like to contribute but I am having trouble following the `DEVELOPER.md`.
First, following all the steps from the guide, it doesn't work on my Mac which I assume is an ARM M1 problem. It says `deepmerge is missing`. Weird!

However, developing it on [Gitpod.io](https://gitpod.io/) seems to work but I am missing steps from the guide.

Right now, I want to just develop in the `studio`. So I do `cd studio && npm install`.
Here I ignored that npm is outputting

```shell
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@supabase/postgres-meta@0.26.1',
npm WARN EBADENGINE   required: { node: '>=14 <15', npm: '>=7' },
npm WARN EBADENGINE   current: { node: 'v16.17.1', npm: '8.15.0' }
npm WARN EBADENGINE }

31 vulnerabilities (30 high, 1 critical)
```

After finishing fetching from the dependencies, I do `npm run dev`.
The UI is appearing but it's stuck loading.

TLDR; I cannot start the studio even following the `DEVELOPER.md`... I know there is a `studio/README.md` but the quickstart is the same and results the same.

What am I missing?

Same when I use yarn...

**Comment by naxola** [comment]
Hi! I have the same issue... and I do not know how may cause that problem:

**error @supabase/postgres-meta@0.26.1: The engine "node" is incompatible with this module. Expected version ">=14 <15". Got "16.18.0"** 

I followed the steps and even in the first part of [documentation ](https://github.com/supabase/supabase/blob/master/DEVELOPERS.md) says **node v16 dependency**.

My npm version is 8.19.2

Any suggestions?

Many thanks
